extensions/pkg/controller/healthcheck/worker/helpers.go

```diff
@@ -39,79 +36,6 @@ func checkMachineDeploymentsHealthy(machineDeployments []machinev1alpha1.Machine
 	return true, nil
 }
 
-func checkNodesScalingUp(machineList *machinev1alpha1.MachineList, readyNodes, desiredMachines int) (gardencorev1beta1.ConditionStatus, error) {
```

Comment:
IIUC, we now drop these checks from the extension library because they are present now in the shoot care controller in gardenlet. But I see that they will be executed only if the MachineControllerManagerDeployment feature gate is enabled.
https://github.com/gardener/gardener/blob/06f3b19981f6ee4fdfb2471b87a1f2cb5bb9f84e/pkg/operation/care/checker.go#L655-L657

I would expect the feature gate to be at least enabled by default (currently alpha and disabled by default) before we drop the duplicate checks in the extension library. Otherwise, if operator uses the default values, these checks will be "turned off" for them when they use a provider extension that vendors this change until the MachineControllerManagerDeployment feature gate is enabled.

Reply:
Good catch, I actually forgot that I did it this way 😅 Will check

Reply:
Please check again, I reverted the commit that dropped this and adapted the confusing TODO statements.